### DISCUSSION TITLE
Relationship data auto save, Halting not working

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6019] [Help]
I write a query to check whether the same form data submitted is already available or not in DB, and it is working as expected in create modal.

```
Actions\CreateAction::make()
    ->before(function (CreateAction $action) {
        ... // my logic
    }),
```

I try to follow the same in edit (modal), but the problem comes here. 

In my form i have this relationship data

```
Forms\Components\Select::make('currency_id')
    ->label(__('message.currency'))
    ->relationship('currency', 'name')
    ->preload()
    ->searchable()
    ->required(),
```
and  below code inside edit actions

```
Tables\Actions\EditAction::make()
    ->before(function (Model $record, array $data, EditAction $action) {
        dd($data);
    }),
```

I can see all my data in dd() after form submit, **BUT currency_id is saved** in database and other form data (like input->text) is not saved. All my logic is also working except the `relationship field` (always store *currency_id* if my logic passed or failed) If i just put `dd()` inside `before()` like my code above, the `relationship field *currency_id*` is always updated. Any idea to prevent this?

Thanks in advance

**Comment by danharrin** [comment]
Try `beforeFormValidated()` instead of `before()`

**Reply by valpuia** [reply]
Same problem

**Reply by danharrin** [reply]
Hm that should work. I'm not sure of the solution then.
